stl/inc/format

```diff
@@ -1248,15 +1243,426 @@ _OutputIt _Write(_OutputIt _Out, basic_string_view<_CharT> _Value) {
     return _STD copy(_Value.begin(), _Value.end(), _Out);
 }
 
-// TODO: placeholder write
-template <class _CharT, class _OutputIt, class _Ty>
-_OutputIt _Write(_OutputIt _Out, _Ty _Val, _Basic_format_specs<_CharT>& _Specs) {
+template <class _CharT, class _OutputIt, class _Func>
+_OutputIt _Write_aligned(
+    _OutputIt _Out, int _Width, _Basic_format_specs<_CharT>& _Specs, _Align _Default_align, _Func&& _Fn) {
+    int _Fill_left  = 0;
+    int _Fill_right = 0;
+
+    if (_Specs._Alignment == _Align::_None)
+        _Specs._Alignment = _Default_align;
+
+    if (_Width < _Specs._Width) {
+        switch (_Specs._Alignment) {
+        case _Align::_Left:
+            _Fill_right = _Specs._Width - _Width;
+            break;
+        case _Align::_Right:
+            _Fill_left = _Specs._Width - _Width;
+            break;
+        case _Align::_Center:
+            _Fill_left  = (_Specs._Width - _Width) / 2;
+            _Fill_right = _Specs._Width - _Width - _Fill_left;
+            break;
+        case _Align::_None:
+        default:
+            _STL_ASSERT(false, "Invalid alignment");
+        }
+    }
+
+    // TRANSITION, add support for unicode/wide formats
+    _Out = _STD fill_n(_Out, _Fill_left, _Specs._Fill[0]);
+    _Out = _Fn(_Out);
+    return _STD fill_n(_Out, _Fill_right, _Specs._Fill[0]);
+}
+
+template <integral _Integral>
+string_view _Get_integral_prefix(char _Type, _Integral _Value) {
+    switch (_Type) {
+    case 'b':
+        return "0b"sv;
+    case 'B':
+        return "0B"sv;
+    case 'x':
+        return "0x"sv;
+    case 'X':
+        return "0X"sv;
+    case 'o':
+        if (_Value != _Integral{0})
+            return "0"sv;
+    }
+    return {};
+}
+
+template <class _OutputIt>
+_OutputIt _Write_sign(_OutputIt _Out, _Sign _Sgn, bool _Is_negative) {
+    if (_Is_negative) {
+        _Out = '-';
+    } else {
+        switch (_Sgn) {
+        case _Sign::_Plus:
+            _Out = '+';
+            break;
+
+        case _Sign::_Space:
+            _Out = ' ';
+            break;
+        }
+    }
+    return ++_Out;
+}
+
+inline void _Fmt_buffer_to_uppercase(char* _Begin, const char* _End) {
+    for (auto _It = _Begin; _It < _End; ++_It) {
+        *_It = static_cast<char>(_CSTD toupper(*_It));
+    }
+}
+
+template <class _CharT, class _Ty>
+bool _In_bounds(_Ty _Value) {

```

Comment:
I think we should decide whether we want to prefix all those helpers with `_Fmt` or get rid of it entirely